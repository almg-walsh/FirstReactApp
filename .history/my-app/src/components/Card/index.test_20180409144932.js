import React from "react";
import { mount } from "enzyme";
import Card from "./index";

describe("Card", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Card />);
    expect(wrapper).toMatchSnapshot();
  });
});
