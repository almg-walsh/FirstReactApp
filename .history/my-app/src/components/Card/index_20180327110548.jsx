import React, { Component } from "react";
import Image from "./Image";
import Button from "./Button";
import Text from "./Text";
import InjectedText from "./InjectedText";

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showTextComponent: false,
      cardNumber: this.props.cardNumber
    };
    this.renderText = this.renderText.bind(this);
  }

  renderText = () => {
    this.setState({
      showTextComponent: true
    });
  };

  render() {
    const { showTextComponent, cardNumber } = this.state;
    return (
      <div className={this.props.className}>
        <Image />
        <Button
          renderText={this.renderText}
          showTextComponent={showTextComponent}
          cardNumber={cardNumber}
        />
        <Text cardNumber={cardNumber} />
        {this.state.showTextComponent ? (
          <InjectedText cardNumber={this.state.cardNumber} />
        ) : null}
      </div>
    );
  }
}

export default Card;