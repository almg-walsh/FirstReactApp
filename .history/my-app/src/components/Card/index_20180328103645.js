import React, { Component } from "react";
import Image from "../Image/index";
import Button from "../Button/index";
import Text from "../Text/index";
import InjectedText from "../InjectedText/index";

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showTextComponent: false,
      cardNumber: this.props.cardNumber
    };
    this.renderText = this.renderText.bind(this);
  }

  renderText = () => {
    this.setState({
      showTextComponent: true
    });
  };

  render() {
    const { showTextComponent, cardNumber } = this.state;
    return (
      <div className={this.props.className}>
        <Image />
        <Button
          renderText={this.renderText}
          showTextComponent={showTextComponent}
          cardNumber={cardNumber}
        />
        <Text cardNumber={cardNumber} />
        {this.state.showTextComponent ? (
          <InjectedText cardNumber={this.state.cardNumber} />
        ) : null}
      </div>
    );
  }
}

export default Card;