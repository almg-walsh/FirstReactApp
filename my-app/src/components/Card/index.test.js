import React from "react";
import { mount } from "enzyme";
import Card from "./index";

describe("Card", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Card />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should show text", () => {
    const wrapper = mount(<Card />);
    const button = wrapper.find("button");
    button.simulate("click", { target: { showTextComponent: true } });
    expect(wrapper.state().showTextComponent).toEqual(true);
  });
});
