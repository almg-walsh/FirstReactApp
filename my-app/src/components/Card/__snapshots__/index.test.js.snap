// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<Card>
  <div>
    <Image>
      <div>
        <img
          alt="logo"
          src="https://avatars0.githubusercontent.com/u/5065156?v=4"
        />
      </div>
    </Image>
    <Button
      renderText={[Function]}
      showTextComponent={false}
    >
      <div>
        <button
          onClick={[Function]}
        >
          Click me
        </button>
      </div>
    </Button>
    <Text>
      <div>
        <p>
          I am the card 
        </p>
      </div>
    </Text>
  </div>
</Card>
`;
